src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -1529,6 +1524,17 @@ FindNodeCheckInRangeTableList(List *rtable, bool (*check)(Node *))
 }
 
 
+/*
+ * QueryContainsDistributedTableRTE determines whether the given
+ * qury contains a distributed table.
+ */
+bool
+QueryContainsDistributedTableRTE(Query *query)
+{
+	return FindNodeCheck((Node *) query, IsDistributedTableRTE);
```

Comment:
I'm wondering whether we should do `FindNodeCheckInRangeTableList(query->rtable, IsDistributedTableRTE)` here, like we do in other places.

Reply:
That's a good point. With our current implementation, we'd probably be fine given that we don't allow `SELECT * FROM ref WHERE x in (SELECT a FROM dist)`. 

But, I'd slightly prefer to keep it, for now, to prevent getting into tricky bugs in the near future?